catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/transformer/CswRecordMapperFilterVisitor.java

```diff
@@ -577,6 +577,9 @@ private double getDistanceInMeters(double distance, String units) {
     return new Distance(distance, linearUnit).getAs(LinearUnit.METER);
   }
 
+  @SuppressWarnings(
+      "squid:S2177" /* This method doesn't and isn't supposed to override it's parent;
+                    it s a copy of a parent function that wasn't inherited */)
   protected Expression visit(Expression expression, Object extraData) {
```

Comment:
It is missing its `@Override` annotation, I put it there now.

However SonarQube still wrongly thinks the method signature `protected Expression visit(Expression expression, Object extraData)` is trying to overwrite a parent method, when in reality that method is copying a parent method that was not passed down.

Reply:
Gotcha. That's...unfortunate and it would be best not to have a method in this class with the same signature as a package-private method in its parent. However...we're probably stuck with it, so the suppression is probably going to have to stay.